client/network/sync/src/lib.rs

```diff
@@ -1031,6 +1035,40 @@ where
 							Vec::new()
 						}
 					},
+					PeerSyncState::DownloadingWarpTarget => {
+						peer.state = PeerSyncState::Available;
+						if let Some(warp_sync) = &mut self.warp_sync {
+							if blocks.len() == 1 {
```

Comment:
Thanks, tried. It doesn't really work out, because we have to check that `blocks` are not empty before calling `validate_blocks()` or its error will be misleading, and we can't cleanly call `validate_blocks()` in the match arm with single block, because it needs a vector.

Reply:
Alright, sad to haer